src/components/common/LangDropdown.jsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import lang from "../images/lang.png";
+import "../index.css";
+
+function LangDropdown() {
+  const { t, i18n } = useTranslation();
+
+  const handleChange = (e) => {
+    switch (e.target.innerText) {
```

Comment:
this switch should not be based off innertext, already you can see a bit of a scaling issue with the code. for every language you support you have to add a case statement! :) 

instead, it should be language-agnostic. you do this by using the value of the dropdown (https://www.w3schools.com/tags/att_option_value.asp) which is the same regardless of language

Reply:
or it seems like you're not using a select for this case but the same thing stands. see my comment below

Reply:
i tried e.target.value but it gave me only a 0. so this was the better but not a good solution i got up with but i will look into that!

Reply:
> i tried e.target.value but it gave me only a 0. so this was the better but not a good solution i got up with but i will look into that!

I think this may solve the issue:
====================
<ul className="lang">
    <li className="lang" onClick={() => i18n.changeLanguage("en")}>
        {t("lang.en")}
    </li>
    <li className="lang" onClick={() => i18n.changeLanguage("tr")}>
        {t("lang.tr")}
    </li>
    <li className="lang" onClick={() => i18n.changeLanguage("ar")}>
        {t("lang.ar")}
    </li>
</ul>

Reply:
@louisrli what do you think about maher's solution? i think it looks good. using functions is good but this is an alternative.